hudi-utilities/src/test/java/org/apache/hudi/utilities/TestHoodieMetadataTableValidator.java

```diff
@@ -183,6 +190,69 @@ public void testMetadataTableValidation(String viewStorageTypeForFSListing, Stri
     assertTrue(validator.getThrowables().isEmpty());
   }
 
+  @Test
+  public void testSecondaryIndexValidation() throws IOException {
+    // To overwrite the table properties created during test setup
+    storage.deleteDirectory(metaClient.getBasePath());
+
+    sparkSession.sql(
+        "create table tbl ("
+            + "ts bigint, "
+            + "record_key_col string, "
+            + "not_record_key_col string, "
+            + "partition_key_col string "
+            + ") using hudi "
+            + "options ("
+            + "primaryKey = 'record_key_col', "
+            + "type = 'mor', "
+            + "hoodie.metadata.enable = 'true', "
+            + "hoodie.metadata.record.index.enable = 'true', "
+            + "hoodie.datasource.write.recordkey.field = 'record_key_col', "
+            + "hoodie.enable.data.skipping = 'true', "
+            + "hoodie.datasource.write.precombine.field = 'ts'"
+            + ") "
+            + "partitioned by(partition_key_col) "
+            + "location '" + basePath + "'");
+
+    Dataset<Row> rows = getRowDataset(1, "row1", "abc", "p1");
+    rows.write().format("hudi").mode(SaveMode.Append).save(basePath);
+    rows = getRowDataset(2, "row2", "cde", "p2");
+    rows.write().format("hudi").mode(SaveMode.Append).save(basePath);
+    rows = getRowDataset(3, "row3", "def", "p2");
+    rows.write().format("hudi").mode(SaveMode.Append).save(basePath);
+
+    // create secondary index
+    sparkSession.sql("create index idx_not_record_key_col on tbl using secondary_index(not_record_key_col)");
+    validateSecondaryIndex();
+
+    // updating record `not_record_key_col` column from `abc` to `cde`
+    rows = getRowDataset(1, "row1", "cde", "p1");
+    rows.write().format("hudi")
+        .option("hoodie.metadata.enable", "true")
+        .option("hoodie.metadata.record.index.enable", "true")
+        .option("hoodie.enable.data.skipping", "true")
+        .option("hoodie.datasource.write.recordkey.field", "record_key_col")
+        .option("hoodie.table.partition.fields", "partition_key_col")
+        .option("hoodie.record.merge.mode", "EVENT_TIME_ORDERING")
```

Comment:
is this a required config now?

Reply:
Addressed. Not required. This was added due to a bug I was hitting.